### DISCUSSION TITLE
update 3.57 version to 4.45 version

### DISCUSSION THREAD

**Discussion by aliramezani747** [LiveHelperChat/livehelperchat#2133] [General]
Hi dears
I'm using 3.57 version and now i want to update to the last version without any change or lose my data.
Do you have any recommendation for updating?
can i update directly to the last version?

**Comment by remdex** [comment]
Read docs about upgrading. Make sure you have 8.2 php

**Comment by aliramezani747** [comment]
Hi remdex
My PHP version is 8.2 but when i installed i get this error.
I searched and check some ways but don't be fix
****************************************************************************************************
Make sure that "cache/*" is writable and then [try again](http://172.28.246.101/index.php/site_admin/install/install)
PDOException Object
(
    [message:protected] => SQLSTATE[42000]: Syntax error or access violation: 1142 CREATE command denied to user 'chat'@'localhost' for table `chat`.`lh_chat`
    [string:Exception:private] => 
    [code:protected] => 42000
    [file:protected] => /var/www/chat/modules/lhinstall/install.php
    [line:protected] => 275
    [trace:Exception:private] => Array
        (
            [0] => Array
                (
                    [file] => /var/www/chat/modules/lhinstall/install.php
                    [line] => 275
                    [function] => query
                    [class] => PDO
                    [type] => ->
                )

            [1] => Array
                (
                    [file] => /var/www/chat/lib/core/lhcore/lhmodule.php
                    [line] => 106
                    [args] => Array
                        (
                            [0] => /var/www/chat/modules/lhinstall/install.php
                        )

                    [function] => include
                )

            [2] => Array
                (
                    [file] => /var/www/chat/lib/core/lhcore/lhmodule.php
                    [line] => 626
                    [function] => runModule
                    [class] => erLhcoreClassModule
                    [type] => ::
                )

            [3] => Array
                (
                    [file] => /var/www/chat/index.php
                    [line] => 50
                    [function] => moduleInit
                    [class] => erLhcoreClassModule
                    [type] => ::
                )

        )

    [previous:Exception:private] => 
    [errorInfo] => Array
        (
            [0] => 42000
            [1] => 1142
            [2] => CREATE command denied to user 'chat'@'localhost' for table `chat`.`lh_chat`
        )

)

**Reply by remdex** [reply]
I ques you don't understand English as error tells already that you db use can't create tables...

**Reply by aliramezani747** [reply]
thank a lot
everything is ok when i use new database for installation but I get 1050 (table already exist) error when i use my old database for restoring my data when install new version.
I searched a lot but didn't fix.

PDOException Object
(
    [message:protected] => SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'lh_chat_blocked_user' already exists
    [string:Exception:private] => 
    [code:protected] => 42S01
    [file:protected] => /var/www/chat/modules/lhinstall/install.php
    [line:protected] => 366
    [trace:Exception:private] => Array
        (
            [0] => Array
                (
                    [file] => /var/www/chat/modules/lhinstall/install.php
                    [line] => 366
                    [function] => query
                    [class] => PDO
                    [type] => ->
                )
            [1] => Array
                (
                    [file] => /var/www/chat/lib/core/lhcore/lhmodule.php
                    [line] => 106
                    [args] => Array
                        (
                            [0] => /var/www/chat/modules/lhinstall/install.php
                        )
                    [function] => include
                )
            [2] => Array
                (
                    [file] => /var/www/chat/lib/core/lhcore/lhmodule.php
                    [line] => 626
                    [function] => runModule
                    [class] => erLhcoreClassModule
                    [type] => ::
                )
            [3] => Array
                (
                    [file] => /var/www/chat/index.php
                    [line] => 50
                    [function] => moduleInit
                    [class] => erLhcoreClassModule
                    [type] => ::
                )
        )
    [previous:Exception:private] => 
    [errorInfo] => Array
        (
            [0] => 42S01
            [1] => 1050
            [2] => Table 'lh_chat_blocked_user' already exists
        )
)

**Reply by remdex** [reply]
https://doc.livehelperchat.com/docs/upgrading
